### DISCUSSION TITLE
cd-ing into a particular directory (scriptability)

### DISCUSSION THREAD

**Discussion by milke** [kovidgoyal/kitty#5044] [Q&A]
It may happen I somehow missed it, searching through discussions and issues I couldn't fine any relevant information, so I'll post it here… Apologies if it's a duplicate.

I've read the thorough [article on scriptability](https://sw.kovidgoyal.net/kitty/remote-control), bit as far as I can see, it covers only ways to control kitty from itself, using `kitty` command. What I need is some basic control from _"the outside"_, e.g. executing a shell script. I need to open a new window and/or a new tab in the front window and then position that new window/tab to a particular directory (`cd` would work, but some other mean, which won't affect terminal history, would be preferable). I can open new window/tab using AppleScript and simulating menu item clicks, but I don't know how I can do something equivalent to `cd`.

Any help or suggestion is greatly appreciated. Thanks.

**Comment by kovidgoyal** [comment]
remote control works from outside kitty too if you specify a socket to listen on. Read that article a little more carefully.

**Reply by milke** [reply]
Right… But to my understanding, for that to work I need to set `allow_remote_control` in `kitty.conf` and I'm looking for a general solution, which wouldn't require a kitty user to change configuration file(s). I can set `allow_remote_control` by executing `kitty -o`, but then I need already running instance of kitty. Or am I not getting the whole remote control process right?

**Comment by kovidgoyal** [comment]
On Sun, May 01, 2022 at 04:46:16AM -0700, Dragan Milić wrote:
> Right… But to my understanding, for that to work I need to set `allow_remote_control` in `kitty.conf` and I'm looking for a general solution, which wouldn't require a kitty user to change configuration file(s). I can set `allow_remote_control` by executing `kitty -o`, but then I need already running instance of kitty. Or am I not getting the whole remote control process right?


Yes, if you want other programs to be able to control kitty, you have to
get the user to allow it.
